trino/sqlalchemy/dialect.py

```diff
@@ -262,19 +262,26 @@ def get_check_constraints(
         return []
 
     def get_table_comment(self, connection: Connection, table_name: str, schema: str = None, **kw) -> Dict[str, Any]:
-        schema = schema or self._get_default_schema_name(connection)
-        if schema is None:
+        catalog_name = self._get_default_catalog_name(connection)
+        if catalog_name is None:
+            raise exc.NoSuchTableError("catalog is required in connection")
```

Comment:
Probably also update `raise exc.NoSuchTableError("schema is required")` -> `raise exc.NoSuchTableError("schema is required in connection")`?

Reply:
@hashhar `schema` can be specified in either param or connection URL, while only specified in connection URL